### PR TITLE
Update heavy-ion eras for 2024 + add MC RelVal workflows

### PR DESCRIPTION
**Pull request by mandrenguyen**
Sets up heavy-ion eras for 2024:
- Implements HI reco, i.e.,`pp_on_PbPb` type eras and their corresponding MC relval workflows (161,161.2,161.02, etc)
- Updates the UPC eras to be based on `Run3_2024`, rather than just `Run3`, although I think for the moment they're equivalent

Note that certain sim customizations for the ZDC and HF fine-grained bit are for now just copied from the 2023 era.  They are extended to the UPC era, although this era is so far only used for the reco step, not for the digi one.  In case the UPC era is used for digi at some point, it's better to change this now.  

The heavy-ion embedding (event overlay) workflows are not extended to 2024 yet.   This will be done in a subsequent PR.  

No backport will be needed.

### CODE REVIEW

**Comment by mandrenguyen** [comment]
test parameters:
   * workflows = 161, 161.02, 161.03, 161.1, 180, 180.1, 181, 181.1, 142.901, 142.902

**Comment by mandrenguyen** [comment]
please test
These tests seem to run file locally.  I'm not sure what this poisoned plugin error mentioned in the logs is about.
For lack of a better idea, I'm just gonna try to the tests again and see if this was transient. 
If anyone has some insight please enlighten me.

**Comment by mmusich** [comment]
>  For lack of a better idea, I'm just gonna try to the tests again and see if this was transient.

this issue was caused by a severely broken IB `CMSSW_14_1_X_2024-06-25-2300` due to [this](https://github.com/cms-sw/cmssw/pull/45037) PR. See discussion there for a rationale. It was fixed in `CMSSW_14_1_X_2024-06-26-1200`

**Comment by mandrenguyen** [comment]
please abort

**Comment by mandrenguyen** [comment]
please test
Thanks @stahlleiton for spotting a typo, which is fixed now

**Comment by mandrenguyen** [comment]
please test

**Comment by mandrenguyen** [comment]
@cms-sw/pdmv-l2 can you have look please?  This one mainly concerns relval.  It would be good to get it in for the next pre-release

**Comment by mandrenguyen** [comment]
@cms-sw/pdmv-l2 @cms-sw/l1-l2 @cms-sw/simulation-l2 @cms-sw/upgrade-l2 
Kindly have a look.

**Comment by aloeliger** [comment]
+l1
- No real changes to L1, just introducing additional era modifiers.

**Comment by civanch** [comment]
+1

**Comment by mandrenguyen** [comment]
ping @cms-sw/pdmv-l2 @cms-sw/upgrade-l2

**Comment by srimanob** [comment]
+Upgrade

**Comment by srimanob** [comment]
@cmsbuild please test

Retrigger the test after 2 weeks, before merging.

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by rappoccio** [comment]
+1

File: Configuration/Eras/python/Era_Run3_pp_on_PbPb_approxSiStripClusters_2024_cff.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import FWCore.ParameterSet.Config as cms
+
+from Configuration.Eras.Era_Run3_2024_cff import Run3_2024
+from Configuration.ProcessModifiers.pp_on_AA_cff import pp_on_AA 
+from Configuration.ProcessModifiers.approxSiStripClusters_cff import approxSiStripClusters
+from Configuration.Eras.Modifier_pp_on_PbPb_run3_cff import pp_on_PbPb_run3
```
Comment (mmusich) [inline Configuration/Eras/python/Era_Run3_pp_on_PbPb_approxSiStripClusters_2024_cff.py:6]: for my education is there a reason why this is using `pp_on_PbPb_run3` as opposed to `pp_on_PbPb_run3_2024` (used in `Configuration/Eras/python/Era_Run3_pp_on_PbPb_2024_cff.py`)?

**Pull request merged**